### PR TITLE
User authentication workflow setup

### PR DESCRIPTION
**Pull request by tonioshikanlu**

### CODE REVIEW

File: Donation-platform/src/main/java/com/google/servlets/UserNameServlet.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import java.io.PrintWriter;
+import java.security.GeneralSecurityException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/* Servlet that returns user information when logged in.*/
+@WebServlet("/username")
+public class UserNameServlet extends HttpServlet
```
Comment (charu0291) [inline Donation-platform/src/main/java/com/google/servlets/UserNameServlet.java:54]: rename this is to GetUserDetailsServlet

File: Donation-platform/src/main/java/com/google/servlets/UserNameServlet.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                }
+                Gson gson = new Gson();
+                String json = gson.toJson(dbCalls.setUsersClass(entity));
+                response.setContentType("application/json;");
+                response.getWriter().println(json);
+
+            }
+            else
+            {
+                System.out.println("Invalid ID token.");
```
Comment (charu0291) [inline Donation-platform/src/main/java/com/google/servlets/UserNameServlet.java:97]: is this for debugging purpose/

Comment (tonioshikanlu) [reply Donation-platform/src/main/java/com/google/servlets/UserNameServlet.java:97]: yes for us to see in the client side console.

File: Donation-platform/src/main/webapp/userAuth.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Gets the user logged in information 
+ * which has been reported by the server and displays user name.
+ */
+function onSignIn(googleUser) {
```
Comment (charu0291) [inline Donation-platform/src/main/webapp/userAuth.js:19]: please rename to more meaningful name like fetchSignInUserInfo

File: Donation-platform/src/main/webapp/userAuth.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    var id_token = googleUser.getAuthResponse().id_token;
+    fetch('/username?idtoken=' + id_token).then(response => response.json()).then((object) => {
+        document.getElementById('index-container').innerHTML = "Welcome: " + object["userName"];
+    });
+}
+/**
+ * Signs the user out of the app.
+ */
+function signOut() {
```
Comment (charu0291) [inline Donation-platform/src/main/webapp/userAuth.js:28]: rename

Comment (tonioshikanlu) [reply Donation-platform/src/main/webapp/userAuth.js]: resolved

File: Donation-platform/src/main/webapp/userAuth.js (hunk 3 above)
Comment (tonioshikanlu) [reply Donation-platform/src/main/webapp/userAuth.js]: resolved

File: Donation-platform/src/main/java/com/google/servlets/UserNameServlet.java (hunk 1 above)
Comment (tonioshikanlu) [reply Donation-platform/src/main/java/com/google/servlets/UserNameServlet.java]: resolved

**Pull request merged**